app/controllers/tool_groups_controller.rb

```diff
@@ -0,0 +1,45 @@
+class ToolGroupsController < ApplicationController
+  before_action :authorize!
+
+  def index
+    render json: ToolGroup.all.order(name: :asc), status: :ok
+  end
+
+  def create
+    create_tool_group
+  rescue ActiveRecord::RecordInvalid
+    update_tool_group
```

Comment:
I don't think we want to rescue and update an existing record if creating it fails. we should return the error that occurred when trying to create it (e.g. missing attribute)

Reply:
I just encountered this issue when trying to create a tool-group with a mis-named attribute, the error I got back was that it couldn't find a tool group with the specified id, which was not the actual problem I had.

Reply:
I was working in this when you commented it. I think now it is just returning the error message and not trying to rescue.

I'll mar it as resolved